cmd/org/context.go

```diff
@@ -0,0 +1,205 @@
+	Version               string
+	ID                    string
+	CloudProvider         string
+	Plan                  string
+	NodeCount             float64
+	ServiceLogs           []sl.ServiceLogShort
+	PdAlerts              map[string][]pd.Incident
+	JiraIssues            []jira.Issue
+	LimitedSupportReasons []*cmv1.LimitedSupportReason
+}
+
+var contextCmd = &cobra.Command{
+	Use:   "context orgId",
+	Short: "fetches information about the given organization",
+	Long: `Fetches information about the given organization. This data is presented as a table where each row includes the name, version, ID, cloud provider, and plan for the cluster.
+Rows will also include the number of recent service logs, active PD Alerts, Jira Issues, and limited support status for that specific cluster.`,
+	Example: `# Get context data for a cluster
+osdctl org context 1a2B3c4DefghIjkLMNOpQrSTUV5`,
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		clusterInfos, err := Context(args[0])
+		if len(clusterInfos) == 0 {
+			return nil
+		}
+
+		printContext(clusterInfos)
+		return err
+	},
+}
+
+func printContext(clusterInfos []ClusterInfo) {
+	table := printer.NewTablePrinter(os.Stdout, 0, 1, 3, ' ')
+	table.AddRow([]string{"DISPLAY NAME", "CLUSTER ID", "VERSION", "STATUS", "PROVIDER", "PLAN", "NODE COUNT", "RECENT SLs", "ACTIVE PDs", "OHSS"})
+
+	for _, clusterInfo := range clusterInfos {
+		status := "Fully Supported"
+		if len(clusterInfo.LimitedSupportReasons) > 0 {
+			status = "Limited Support"
+		}
+
+		plan := clusterInfo.Plan
+		if plan == "MOA" {
+			plan = "ROSA"
+		}
+		if plan == "MOA-HostedControlPlane" {
+			plan = "HCP"
+		}
+
+		recentSLs := len(clusterInfo.ServiceLogs)
+		activePDs := len(clusterInfo.PdAlerts)
+		ohss := len(clusterInfo.JiraIssues)
+		table.AddRow([]string{
+			clusterInfo.Name,
+			clusterInfo.ID,
+			clusterInfo.Version,
+			status,
+			clusterInfo.CloudProvider,
+			plan,
+			fmt.Sprintf("%v", clusterInfo.NodeCount),
+			strconv.Itoa(recentSLs),
+			strconv.Itoa(activePDs),
+			strconv.Itoa(ohss),
+		})
+	}
+
+	table.AddRow([]string{})
+	table.Flush()
+}
+
+func Context(orgId string) ([]ClusterInfo, error) {
+	clusterSubscriptions, err := SearchAllSubscriptionsByOrg(orgId, statusActive, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch cluster subscriptions for org with ID %s: %w", orgId, err)
+	}
+
+	clusterSubscriptionsCount := len(clusterSubscriptions)
+
+	if clusterSubscriptionsCount == 0 {
+		return nil, nil
+	}
+
+	// cluster info
+	ocmClient, err := utils.CreateConnection()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OCM client: %w", err)
+	}
+	defer ocmClient.Close()
+
+	var orgClustersInfo []ClusterInfo
+
+	var wg sync.WaitGroup
+	var mutex sync.Mutex
+	count := 0
+
+	fmt.Printf("Fetching data for %v clusters in org %v...\n", clusterSubscriptionsCount, orgId)
+	fmt.Printf("Fetched data for 0 of %v clusters...\n", clusterSubscriptionsCount)
```

Comment:
1. Can you print these to stderr so I don't have to filter them out?
1. Was the "0 of n" from testing? I'm just not sure of the value with the previous log.

Reply:
If you look down below at lines 193 through 198, this line will update in-place as data is fetched. We could have it start with nothing instead of "0 of n" if we'd like, but I thought this looked better when I tested both in my console.

Reply:
As for printing those results to stderr, I'm not sure that makes sense as these aren't error messages.